src/beautifiers/index.coffee

```diff
@@ -241,15 +240,19 @@ module.exports = class Beautifiers extends EventEmitter
         logger.info('beautify', text, allOptions, grammar, filePath, onSave)
         logger.verbose(allOptions)
 
-        # Get language
-        fileExtension = path.extname(filePath)
-        # Remove prefix "." (period) in fileExtension
-        fileExtension = fileExtension.substr(1)
-        languages = @languages.getLanguages({grammar, extension: fileExtension})
-        logger.verbose(languages, grammar, fileExtension)
+        languages = []
+        fileExtension = null
+        getLanguages = (filePath) =>
```

Comment:
I'd prefer if `getLanguage(grammar, filePath)` was changed to `getLanguages(grammar, filePath)` and then a small function `getLanguage(grammar, filePath)` was added which would call `getLanguages` and return the first result. Therefore it would be backwards compatible (has getLanguage(grammar, filePath)` ), no duplicated code, and no function defined within scope of `getOptionsForLanguage`.